src/vm/functions/principals.rs

```diff
@@ -45,3 +48,76 @@ pub fn special_is_standard(
         (address_is_mainnet && context_is_mainnet) || (address_is_testnet && !context_is_mainnet),
     ))
 }
+
+pub fn special_parse_principal(
+    args: &[SymbolicExpression],
+    env: &mut Environment,
+    context: &LocalContext,
+) -> Result<Value> {
+    check_argument_count(2, args)?;
+    runtime_cost(ClarityCostFunction::Unimplemented, env, 0)?;
+
+    // Handle the block property name input arg.
+    let property_name = args[0]
+        .match_atom()
+        .ok_or(CheckErrors::ParsePrincipalExpectPropertyName)?;
+
+    let principal_property = PrincipalProperty::lookup_by_name(property_name).ok_or(
+        CheckErrors::NoSuchParsePrincipalProperty(property_name.to_string()),
```

Comment:
`version` and `hashbytes` sounds more like components than properties to me. What do you think?

Reply:
What's teh difference between an "component" and a "property"?

What code changes are you suggesting?

Reply:
What's your take shifting gears a bit here, and instead of going with `(principal-parse <something> principal) -> (buff 1) || (buff 20)`, go with something like `(principal-destruct principal) -> { version: (buff 1), hashbytes: (buff 20) }` ?
That way we don't have to do this lookup, and introduce a new error variant / pitfall for developers.